src/zenml/integrations/seldon/steps/seldon_deployer.py

```diff
@@ -192,3 +241,151 @@ def prepare_service_config(model_uri: str) -> SeldonDeploymentConfig:
+            f" meet the criteria. Reusing the last model server deployed by step "
+            f"'{step_name}' and pipeline '{pipeline_name}' for model "
+            f"'{config.service_config.model_name}'..."
+        )
+        service = cast(SeldonDeploymentService, existing_services[0])
+        # even when the deployment decision is negative, we still need to start
+        # the previous model server if it is no longer running, to ensure that
+        # a model server is available at all times
+        if not service.is_running:
+            service.start(timeout=config.timeout)
+        return service
+
+    # entrypoint for starting Seldon microservice deployment for custom model
+    entrypoint_command = [
+        "python",
+        "-m",
+        "zenml.integrations.seldon.custom_deployer.zenml_custom_model",
+        "--model_name",
+        config.service_config.model_name,
+        "--predict_func",
+        config.custom_deploy_parameters.predict_function,
+    ]
+
+    # verify if there is an active stack before starting the service
+    if not context.stack:
+        raise DoesNotExistException(
+            "No active stack is available. "
+            "Please make sure that you have registered and set a stack."
+        )
+    stack = context.stack
+
+    # prepare the custom deployment docker image
+    custom_docker_image_name = model_deployer.prepare_custom_deployment_image(
+        pipeline_name=pipeline_name,
+        stack=stack,
+        docker_configuration=docker_configuration,
+        runtime_configuration=runtime_configuration,
+        entrypoint=entrypoint_command,
+    )
+
+    # copy the model files to new specific directory for the deployment
+    served_model_uri = os.path.join(context.get_output_artifact_uri(), "seldon")
+    fileio.makedirs(served_model_uri)
+    io_utils.copy_dir(model.uri, served_model_uri)
+
+    # Get the model artifact to extract information about the model
+    # and how it can be loaded again later in the deployment environment.
+    artifact = stack.metadata_store.store.get_artifacts_by_uri(model.uri)
+    if not artifact:
+        raise DoesNotExistException("No artifact found at {}".format(model.uri))
+
+    # save the model artifact metadata to the YAML file and copy it to the
+    # deployment directory
+    model_metadata_file = save_model_metadata(artifact[0])
+    fileio.copy(
+        model_metadata_file,
+        os.path.join(served_model_uri, MODEL_METADATA_YAML_FILE_NAME),
+    )
+
+    # prepare the service configuration for the deployment
+    service_config = config.service_config.copy()
+    service_config.model_uri = served_model_uri
+
+    # create the specification for the custom deployment
+    service_config.spec = create_seldon_core_custom_spec(
+        model_uri=service_config.model_uri,
+        custom_docker_image=custom_docker_image_name,
+        secret_name=model_deployer.kubernetes_secret_name,
```

Comment:
This secret contains credentials that allow the seldon predictor containers to authenticate to the remote object storage service (S3, GCS, etc.). What you need here is a secret that contains credentials that allow the kube cluster to authenticate to the container registry.